src/Behat/Config/Formatter/ProgressFormatter.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Behat\Config\Formatter;
+
+final class ProgressFormatter extends Formatter
+{
+    public const NAME = 'progress';
+
+    public function __construct(
+        ?bool $timer = null,
+    ) {
+        $settings = self::defaults();
+
+        $settings['timer'] = $timer ?? $settings['timer'];
```

Comment:
I think I would do this the other way round - leave it as:
```suggestion
        bool $timer = true,
    ) {
        $settings = ['timer' => true]
```

And then implement something like this:

```php
public static function defaults(): array {
  return (new self())->toArray()
}
```

That way there's less scope for errors, it's easy for the user to see what default will be applied, and we can keep nullable typehints for options that are actually nullable?

Reply:
Yes that seems smart. I'll try it soon, thx 👍

Reply:
done